replay_build_scripts/build-branch-pipeline.mjs

```diff
@@ -0,0 +1,55 @@
+import { load } from "js-yaml";
+import { promises as fs } from "fs";
+
+import { spawnChecked } from "./common.mjs";
+
+async function getBuildkiteMetadata(key) {
+  // buildkite-agent meta-data get <key>
+  const result = spawnChecked("buildkite-agent", ["meta-data", "get", key]);
+  const output = result.stdout.toString().trim();
+
+  return output;
+}
+
+async function getConfiguredPlatformArchitectures() {
+  const output = await getBuildkiteMetadata("platform-architecture");
+  const platformArchitectures = output.split("\n");
+
+  return platformArchitectures;
+}
+
+async function main() {
+  const yamlString = await fs.readFile(".buildkite/pipeline.yml", "utf8");
+  const parsedYaml = load(yamlString);
+  const targetPlatformArchitectures =
+    await getConfiguredPlatformArchitectures();
+  const buildStepKeys = targetPlatformArchitectures.map(
+    (platformArchitecture) => `build-chromium-${platformArchitecture}`
+  );
+
+  const yamlWithoutExcludedBuildSteps = parsedYaml.steps.filter((step) => {
+    if (
+      (step.key &&
+        step.key.startsWith("build-") &&
+        !buildStepKeys.includes(step.key)) ||
+      buildStepKeys.includes(step.depends_on)
```

Comment:
I'm provisionally okay with this change, though I reserve the right to reverse my opinion :)  My issue with all kinds of codegen like this is that it takes an annoying-but-simple problem (manual labour of specifying build steps for additional os/architectures) and converts it into a fun-but-complex programming problem (correctly generating the code) that also obscures what we're trying to actually do (build and test some stuff.)  If this starts blowing up, I'm going to just copy-paste the steps myself :)

Reply:
Deal!